internal/yml/merge_test.go

```diff
@@ -380,3 +380,34 @@ e:
 		t.Errorf("Mismatch (-expected +actual):\n%s", diff)
 	}
 }
+
+func TestMergeLockedElementOnlyPresentInTargetAndNotLastOfHisSiblingsPreservesItsOrder(t *testing.T) {
+	aContent := `
+a: b
+c: d
+e:
+  f: g
+  l: m
+`
+	bContent := `
+a: b
+c: d
+e:
+  f: g
+  h: i
+  ## lock
+  j: k
+  l: m
+`
+	expectedContent := `
+a: b
+c: d
+e:
+  f: g
+  ## lock
+  j: k
+  l: m
```

Comment:
Out of curiosity, what was the output before the fix? More or less random?

Reply:
No, it was predictable, always appended at the end. It's just that it would result in Joy wanting to update such nodes just to move them to the end, resulting in unneeded changes.